Source/BugsnagClient.m

```diff
@@ -811,14 +815,26 @@ - (void)orientationChanged:(NSNotification *)notif {
         [orientationNotifName isEqualToString:lastBreadcrumb[BSGKeyName]]) {
         NSDictionary *metadata = lastBreadcrumb[BSGKeyMetadata];
 
-        if ([orientation isEqualToString:metadata[BSGKeyOrientation]]) {
+        if ([orientation isEqualToString:metadata[BSGKeyOrientation]])
             return; // ignore duplicate orientation event
-        }
     }
 
-    [[self state] addAttribute:BSGKeyOrientation
-                     withValue:orientation
-                 toTabWithName:BSGKeyDeviceState];
+    // It's not a change
+    if ([orientation isEqualToString:lastOrientation])
+        return;
+
+    // We previously had an orientation
+    if (lastOrientation) {
```

Comment:
This can be simplified by calculating the `lastOrientation` when `BugsnagClient` is initialised (assuming that the return value of `[UIDevice currentDevice].orientation` is non-null)

Reply:
We can't assume this.  Acording to [the docs](https://developer.apple.com/documentation/uikit/uidevice/1620041-begingeneratingdeviceorientation?language=objc) `beginGeneratingDeviceOrientationNotifications()` needs called.  We do this in `BugsnagClient.start()`.  As the code stands we call `BugsnagClient.initWithConfiguration()` first, then `BugsnagClient.start()`.  We could take the golden path and short-circuit on the first call instead.

Reply:
Isn't it possible to move where `beginGeneratingDeviceOrientationNotifications()` is called to make calculation of an initial value feasible?

Reply:
Yes, but I don't believe that there's any guarantee that the orientation has been determined at any specific time.  A test implementation bears this out.  One way round this deficit would be to use `[[UIApplication sharedApplication] statusBarOrientation]` but this does not allow for face up/down determination.  If the goal is to have a breadcrumb with valid from/to values I think the current approach with an early exit is the most consistent.

Reply:
Happy enough to go with the current implementation if it's not always feasible to get an initial value.